typespec-extension/src/code-model-builder.ts

```diff
@@ -1598,7 +1598,14 @@ export class CodeModelBuilder {
             }
           });
         }
+      } else {
+        pushDistinct(objectSchema.parents.all, parentSchema);
       }
+    } else if (isRecordModelType(this.program, type)) {
```

Comment:
Do we need pure Record check here?

Reply:
No, if it gets here (the `processObjectSchema`), it is already mixed.

pure one already handled above with `processObjectSchema`